lib/iris/common/resolve.py

```diff
@@ -1531,13 +1808,137 @@ def cube(self, data, in_place=False):
+            **higher rank** operand (``tgt``). Therefore, the ``src``
+            :class:`~iris.cube.Cube` may be either the ``lhs`` or the ``rhs``
+            :class:`~iris.cube.Cube` operand, given the direction of the
+            mapping. See :attr:`~iris.common.resolve.Resolve.map_rhs_to_lhs`.
+
+        If no :class:`~iris.cube.Cube` operands have been provided, then
+        ``mapped`` is ``None``.
+
+        For example,
+
+        .. doctest::
+
+            >>> print(cube1)
+            air_temperature / (K)               (time: 240; latitude: 37; longitude: 49)
+                 Dimension coordinates:
+                      time                           x              -              -
+                      latitude                       -              x              -
+                      longitude                      -              -              x
+                 Auxiliary coordinates:
+                      forecast_period                x              -              -
+                 Scalar coordinates:
+                      forecast_reference_time: 1859-09-01 06:00:00
+                      height: 1.5 m
+                 Attributes:
+                      Conventions: CF-1.5
+                      Model scenario: A1B
+                      STASH: m01s03i236
+                      source: Data from Met Office Unified Model 6.05
+                 Cell methods:
+                      mean: time (6 hour)
+            >>> print(cube2)
+            air_temperature / (K)               (longitude: 49; latitude: 37)
+                 Dimension coordinates:
+                      longitude                           x             -
+                      latitude                            -             x
+                 Scalar coordinates:
+                      forecast_period: 10794 hours
+                      forecast_reference_time: 1859-09-01 06:00:00
+                      height: 1.5 m
+                      time: 1860-06-01 00:00:00, bound=(1859-12-01 00:00:00, 1860-12-01 00:00:00)
+                 Attributes:
+                      Conventions: CF-1.5
+                      Model scenario: E1
+                      STASH: m01s03i236
+                      source: Data from Met Office Unified Model 6.05
+                 Cell methods:
+                      mean: time (6 hour)
+            >>> Resolve().mapped is None
+            True
+            >>> resolver = Resolve(cube1, cube2)
+            >>> resolver.mapped
+            True
+            >>> resolver.map_rhs_to_lhs
+            True
+            >>> resolver = Resolve(cube2, cube1)
+            >>> resolver.mapped
+            True
+            >>> resolver.map_rhs_to_lhs
+            False
 
         """
-        return self._src_cube.ndim == len(self.mapping)
+        result = None
+        if self.mapping is not None:
+            result = self._src_cube.ndim == len(self.mapping)
+        return result
```

Comment:
Adds a bit more rigour to deal with the case where the user attempt to determine `mapped`, but no operands have been provisioned to analyse